### PR TITLE
LibDNS+RequestServer: Miscellaneous fixes and cleanup

### PR DESCRIPTION
**Pull request by rmg-x**
See individual commits

### CODE REVIEW

File: Services/RequestServer/ConnectionFromClient.cpp [hunk 1]
```diff
@@ -447,7 +446,6 @@ void ConnectionFromClient::start_request(i32 request_id, ByteString const& metho
             }
 
             auto formatted_address = resolve_opt_builder.to_byte_string();
-            g_dns_cache.set(host, formatted_address);
```
Comment (alimpfard) [inline Services/RequestServer/ConnectionFromClient.cpp:450]: Nononono, this thing is holding on to the strings, without this the next access will do a UAF read

Comment (rmg-x) [reply Services/RequestServer/ConnectionFromClient.cpp:450]: Oh... I'll remove that.

File: Services/RequestServer/ConnectionFromClient.cpp [hunk 2]
```diff
@@ -422,9 +423,10 @@ void ConnectionFromClient::start_request(i32 request_id, ByteString const& metho
             if (did_set_body && !request_headers.contains("Content-Type"))
                 curl_headers = curl_slist_append(curl_headers, "Content-Type:");
 
-            for (auto const& header : request_headers.headers()) {
-                auto header_string = ByteString::formatted("{}: {}", header.name, header.value);
+            for (auto const& [name, value] : request_headers.headers()) {
+                auto header_string = ByteString::formatted("{}: {}", name, value);
                 curl_headers = curl_slist_append(curl_headers, header_string.characters());
+                request->formatted_request_headers.append(move(header_string));
```
Comment (awesomekling) [inline Services/RequestServer/ConnectionFromClient.cpp:429]: According to https://curl.se/libcurl/c/curl_slist_append.html
> [curl_slist_append](https://curl.se/libcurl/c/curl_slist_append.html) copies the string.

So I don't understand what this is fixing? Did you see an ASAN error related to this?

Comment (rmg-x) [reply Services/RequestServer/ConnectionFromClient.cpp]: Ah, you're correct. Removed. 
Someone made an observation on discord in #network but I didn't verify what `curl_slist_append` was actually doing, just assumed.

**Comment by rmg-x** [comment]
@alimpfard I've removed the "g_dns_cache" again, but this time because we know that `curl_slist_append` copies the string. I don't believe we need to keep those around ourselves anymore.

**Comment by alimpfard** [comment]
That's odd, I remember seeing invalid data being read by curl when that hashmap wasn't present (that's actually why I added it to begin with) :thinking:

**Pull request merged**